.github/workflows/release.yml

```diff
@@ -0,0 +1,52 @@
+name: Release or Create Release PR
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  release-or-create-pr:
+    # Only one publish job should run at a time
+    # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency
+    concurrency: publish
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      - name: Setup npm auth token
+        run: |
+          echo -e "npmRegistryServer: \"https://registry.npmjs.org/\"\nnpmAuthToken: \"$NPM_TOKEN\"" >> ./.yarnrc.yml
+        env:
+          NPM_TOKEN: '${{ secrets.NPM_AUTH_TOKEN }}'
+
+      - name: Release or Create Release PR
+        uses: changesets/action@v1
+        id: changesets
+        with:
+          version: yarn version
+          publish: yarn publish
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Publish Docs
```

Comment:
(Not required) You could also make a script and on the `yarn publish` script do `yarn npm publish --tolerate-republish && ./scripts/publish-docs`

Reply:
I had to make some changes to the publish stuff, I will clean up the scripts soon, but for now its best to leave this as is I think.